packages/es-client/search.js

```diff
@@ -65,12 +56,19 @@ const esProdConfig = async (host) => {
     node = `https://${host}`;
   }
 
+  const credentialsProvider = fromNodeProviderChain({
+    clientConfig: {
+      region: process.env.AWS_REGION,
+    },
+  });
+
+  const credentials = await credentialsProvider();
```

Comment:
Does the `credentials` object here look similar/the same as what `getCredentials` was previously returning? Makes sense if so.

Reply:
Oh hmm I see in the mock below that it works a little different. We're using keys instead of user/pass. Are there any implications there? 🤔

Reply:
Good questions and I'm going to @Jkovarik in my responses because he's specifically working on a bug related to expired credentials.

The return type of credentials I believe is _close enough_ to what we were originally getting: 
**New:** https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-smithy-types/TypeAlias/AwsCredentialIdentityProvider/

**Old:** https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html

The biggest difference (and something @Jkovarik might care about) is `expiration` vs `expireTime`. I don't think the previous code actually used `expireTime` and why we have the other bug that we do.

Do we _actually_ support user/pass and was that not just a relic of an easy mock? If so, we might need more than the Node provider chain. The Node provider chain looks at the following for creds (in order):

1. [Environment variables exposed via process.env](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromenv)
2. [SSO credentials from token cache](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromsso)
3. [Web identity token credentials](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromtokenfile)
4. [Shared credentials and config ini files](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromini)
5. [The EC2/ECS Instance Metadata Service](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromcontainermetadata-and-frominstancemetadata)

Reply:
Based on Slack conversations, I think this can be resolved. It appears that the mock was done using user/pass for simplicity rather than an attempt to emulate what actually happens.